### PR TITLE
RPRBLND-1370: Reroute nodes

### PR DESCRIPTION
**Pull request by VascoPi**
### PURPOSE
Add Reroute nodes to MaterialX, Shader/World Editor, USD Editor.
Optionally add Frame node.

### EFFECT OF CHANGE
Added Layout node category to MaterialX, Shader Editor, USD Editor.
Added Reroute and Frame nodes to MaterialX, Shader Editor, USD Editor.

### TECHNICAL STEPS
Registered new category Layout in USD, MaterialX, Shader Editor.
Changed get_input_link method so it skips all NodeReroute objects. It's has done in MxNode, NodeParser, USDNode.
In HDUSD_MATERIAL_PT_material_settings_surface, HDUSD_MATERIAL_PT_material_settings_displacement, changed draw method so it skips all NodeReroute objects.
Changed _reset_next method so it skips all NodeReroute objects in USDNode.

### CODE REVIEW

File: src/hdusd/bl_nodes/node_parser.py [hunk 1]
```diff
@@ -347,6 +347,15 @@ def get_input_link(self, in_key: [str, int]):
...
             return None
 
+        node = link.from_node
+        while isinstance(node, bpy.types.NodeReroute):
+            if not node.inputs[0].links:
+                return None
+
+            link = node.inputs[0].links[0]
+            if link.is_valid:
+                node = link.from_node
```
Comment (bnagirniak) [inline src/hdusd/bl_nodes/node_parser.py:357]: Please move this code to utils/\_\_init__.py
```
    def pass_node_reroute(node)
        while isinstance(node, bpy.types.NodeReroute):
            if not node.inputs[0].links:
                return None

            link = node.inputs[0].links[0]
            if link.is_valid:
                node = link.from_node

        return node
```
and use it in other places

**Review by bnagirniak — CHANGES_REQUESTED** [review]
Suggested code improvement

**Review by DagerD — APPROVED** [review]
Tested. Works ok.
Also commented in code

**Review by bnagirniak — APPROVED** [review]
Tested - works good

**Pull request merged**